### PR TITLE
Split out foundation 

### PR DESCRIPTION
**Pull request by imduffy15**
This pull requests adds an extra step between transitioning from foundation to base.

In the event we're using multiple different hypervisors or cloud providers it would be nice to use a single vmdk created from an ISO installation on these several platforms without rebuilding the disk for each.

The foundation image exists to create a JeOS VMDK. The Membrane exists to layer on-top platform specific tools: vmware tools, virtual box guest additions, cloud-init, walinuxagent etc.

Additionally, i've modified foundation to output a json object that can be used as an input in the membrane box. Using jq we can do a merge of the two variable files:

```
cat <half-filled-variables> <dynamic-variables> | jq add -s -r > variables.json
packer build -var-file variables.json packer.json
``
```

### CODE REVIEW

File: boxes/foundation/www/ks.cfg [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+ca-certificates
+centos-release
+checkpolicy
+chkconfig
+coreutils
+coreutils-libs
+cpio
+cracklib
+cracklib-dicts
+cronie
```
Comment (samdunne) [inline boxes/foundation/www/ks.cfg:68]: Do we want this?

File: scripts/aws.sh [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+cloud_config_modules:
+ - mounts
+ - locale
+ - set-passwords
+ - timezone
+ - yum-add-repo
+ - package-update-upgrade-install
+ - puppet
+ - chef
+ - salt-minion
```
Comment (samdunne) [inline scripts/aws.sh:42]: Why salt and mcollective?

Comment (imduffy15) [reply scripts/aws.sh:42]: Its part of the cloud-init default config.

File: boxes/foundation/www/ks.cfg (hunk 1 above)
Comment (imduffy15) [reply boxes/foundation/www/ks.cfg:68]: Yes. Its part of crontab.

File: scripts/aws.sh (hunk 2 above)
Comment (samdunne) [reply scripts/aws.sh:42]: Neat

File: boxes/membrane/packer.json [hunk 3]
```diff
@@ -4,10 +4,10 @@
       "headless": "{{ user `headless` }}",
       "name": "{{ user `virtualbox_identifier` }}",
       "output_directory": "builds/{{ user `group_id` }}.{{ user `virtualbox_identifier` }}.{{ user `artifact_id` }}/{{ user `major_version` }}.{{ user `minor_version` }}.{{ user `patch_version` }}.{{ user `incremental_version` }}/",
+      "shutdown_command": "/sbin/halt -h -p",
```
Comment (samdunne) [inline boxes/membrane/packer.json:7]: Lexicographical order :(

File: boxes/membrane/variables.json [hunk 4]
```diff
@@ -0,0 +1,14 @@
+{
+  "artifact_id": "centos6-membrane",
+  "disk_size": "61440",
+  "group_id": "com.gauchoconsulting.cloud",
+  "headless": "false",
+  "incremental_version": "0",
+  "major_version": "0",
+  "minor_version": "0",
+  "patch_version": "0",
+  "virtualbox_source_virtualdisk_path": "builds/com.gauchoconsulting.cloud.virtualbox.centos6-foundation/0.0.0.0/com.gauchoconsulting.cloud.virtualbox.centos6-foundation-0.0.0.0-disk1.vmdk",
```
Comment (samdunne) [inline boxes/membrane/variables.json]: Shouldn't this be `builds/com/gauchoconsulting...`?

**Pull request merged**